api/src/main/java/org/pmiops/workbench/cohortbuilder/CohortBuilderServiceImpl.java

```diff
@@ -275,33 +275,14 @@ public List<Criteria> findCriteriaAutoCompleteV2(
 
     SearchTerm searchTerm = new SearchTerm(term, mySQLStopWordsProvider.get().getStopWords());
 
-    List<DbCriteria> criteriaList = new ArrayList<>();
-
-    if (searchTerm.hasModifiedTermOnly()) {
-      criteriaList =
-          cbCriteriaDao.findCriteriaByDomainAndTypeAndStandardAndFullText(
-              domain, type, standard, searchTerm.getModifiedTerm(), pageRequest);
-    } else if (searchTerm.hasEndsWithOnly()) {
-      criteriaList =
-          cbCriteriaDao.findCriteriaByDomainAndTypeAndStandardAndNameEndsWith(
-              domain, type, standard, searchTerm.getEndsWithTerms(), pageRequest);
-    } else if (searchTerm.hasEndsWithTermsAndModifiedTerm()) {
-      criteriaList =
-          cbCriteriaDao.findCriteriaByDomainAndTypeAndStandardAndTermAndNameEndsWith(
-              domain,
-              type,
-              standard,
-              searchTerm.getModifiedTerm(),
-              searchTerm.getEndsWithTerms(),
-              pageRequest);
-    }
+    List<DbCriteria> criteriaList =
+        cbCriteriaDao.findCriteriaAutoComplete(domain, type, standard, searchTerm, pageRequest);
 
+    // find by code if auto complete return nothing.
     if (criteriaList.isEmpty()) {
-      // why we do not use term.replaceAll("[()+\"*-]", "") as in
-      // findCriteriaByDomain()
       criteriaList =
           cbCriteriaDao.findCriteriaByDomainAndTypeAndStandardAndCode(
-              domain, type, standard, term, pageRequest);
+              domain, type, standard, term.replaceAll("[()+\"*-]", ""), pageRequest);
```

Comment:
Your comment was correct, so added this regex and removed the comment.

Reply:
:-)